robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot

```diff
@@ -0,0 +1,93 @@
+*** Settings ***
+
+Resource        robot/Cumulus/resources/NPSP.robot
+Library         cumulusci.robotframework.PageObjects
+...             robot/Cumulus/resources/GiftEntryPageObject.py
+...             robot/Cumulus/resources/OpportunityPageObject.py
+...             robot/Cumulus/resources/PaymentPageObject.py
+...             robot/Cumulus/resources/AccountPageObject.py
+Suite Setup     Run keywords
+...             Open Test Browser
+...             API Check And Enable Gift Entry
+...             Setup Test Data
+Suite Teardown  Capture Screenshot and Delete Records and Close Browser
+
+*** Keywords ***
+Setup Test Data
+    [Documentation]      Creates the organiation account,opportunity and queries the payment record required for the test
+    ...                  along with getting dates and namespace required for test.
+    &{account} =         API Create Organization Account    Name=${faker.company()}
+    Set suite variable   &{ACCOUNT}
+    ${fut_date} =            Get Current Date         result_format=%Y-%m-%d    increment=2 days
+    Set suite variable   ${FUT_DATE}
+    ${cur_date} =            Get Current Date         result_format=%Y-%m-%d    
+    Set suite variable   ${CUR_DATE}
+    &{opportunity} =     API Create Opportunity   ${ACCOUNT}[Id]              Donation  
+    ...                  StageName=Prospecting    
+    ...                  Amount=500    
+    ...                  CloseDate=${FUT_DATE}    
+    ...                  npe01__Do_Not_Automatically_Create_Payment__c=false    
+    ...                  Name=${ACCOUNT}[Name] Donation
+    Set suite variable   &{OPPORTUNITY}
+    ${ui_date} =         Get Current Date                   result_format=%b %-d, %Y
+    Set suite variable   ${UI_DATE}
+    &{payment} =         API Query Record         npe01__OppPayment__c      npe01__Opportunity__c=${OPPORTUNITY}[Id]
+    Set suite variable   &{PAYMENT}
```

Comment:
Did you investigate if we can make this part of the keyword Setupdata?

Reply:
I tried doing this and if I use this it is taking 18 seconds to setup data verses 2 secs so using the straightforward method